### DISCUSSION TITLE
Import color

### DISCUSSION THREAD

**Discussion by stepsolar** [ExperienceLovelace/ha-floorplan#307] [Q&A]
Hello,
would it be possible to import a color code from a link? so as to make itself dynamic based on the color change?

**Comment by exetico** [comment]
Color code from a link?

You'll need to spend a few more minutes to explain your thought.

Right now, we're two steps away from a https://xyproblem.info/ ☺️.

Do us a favor and replain the case, and include what you have, what you need, and what's actually the problem you're trying to resolve. If there's any sub-related problems, share them to.

Do you have the link, and maybe even your SVG? Add that too ☺️. All the things will help us, and other people dropping by this discussion in the future.

**Reply by stepsolar** [reply]
I have an object that needs to change color based on a color on a website. I was thinking of doing a scraping, but it must then be able to integrate into the CSS color. It would be possible?

**Reply by OzGav** [reply]
https://www.home-assistant.io/integrations/scrape/

**Reply by exetico** [reply]
It's possible to scrape in HA, but you could also do this as a python or node script in HA. Another way is to just fetch the data in JS with [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API). You could also fetch things with a 3rd-party solution, and just expose the data from a DB pulled by HA. There's plenty of ways.

But I still think you're asking for a very specific thing, while hidden the real reason/idea/thoughts. I can't see any good reason why a color should be fetch/import/grap a color from another web-resource. The only reason I can think of, is if you're more "suited" to code in another language and pull the results from another endpoint.

Again, if possible: Consider to share what you're actually trying to do here. Scraping data has nothing to do with ha-floorplan or anything related to it, but there must be a (good) reason why you're using our discussion area :smiley:.

I hope you understand. I'm just trying to help you out, and improve the overall quality of the discussion.

**Reply by stepsolar** [reply]
> It's possible to scrape in HA, but you could also do this as a python or node script in HA. Another way is to just fetch the data in JS with [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API). You could also fetch things with a 3rd-party solution, and just expose the data from a DB pulled by HA. There's plenty of ways.
> 
> But I still think you're asking for a very specific thing, while hidden the real reason/idea/thoughts. I can't see any good reason why a color should be fetch/import/grap a color from another web-resource. The only reason I can think of, is if you're more "suited" to code in another language and pull the results from another endpoint.
> 
> Again, if possible: Consider to share what you're actually trying to do here. Scraping data has nothing to do with ha-floorplan or anything related to it, but there must be a (good) reason why you're using our discussion area 😃.
> 
> I hope you understand. I'm just trying to help you out, and improve the overall quality of the discussion.

Thanks to your project I'm making a screen similar to the solardge dashboard showing the solar panels during their production, they change color (not opacity) based on the power that is generated.  Since I would like to make it as similar as possible to thesolardge dashboard I also thought of using the website scraper creating some color sensors and in your CSS insert the color sensor.  Your project for now is the best that fits.  Thank you

**Reply by exetico** [reply]
I see :) In that cause, I'd not recommend to scrape the data - just grap the colors they're working with, and make it as similar as possible. Sadly I don't have a solardge or simular project, so I can't really help on how to get the proper colors. But I'd recommend you to use the inspect panel in your browser and see how the colors is generated (if it's done in the browser end), or else, just grap some of the colors from the panel, and try and "fill in" the rest of the spots.

It does make sense to go with a color, rather than a opacity, if you'd like to get it closer to what you have in the Solardge dashboard :smile: .
